alphagov/paas-rds-metric-collector
# [#158456560] Add a lot of metrics to the collector

What?
-----

We want to add multiple metrics to the collector.
We will add metrics from cloudwatch [1] and SQL postgres ones.

[1] https://docs.aws.amazon.com/AmazonCloudWatch/latest/monitoring/rds-metricscollected.html

In this PR we added multiple metrics we found useful. We were not sure about
some of the cloudwatch ones.

We additionally added the capability of tagging the metrics, which would allow us to tag metrics with dynamic names like `table_name`

We did not implement metrics for query latency or slow queries, as that seems
more complicated: We need to enable slow query logging in postgres parameter
groups and read the logs using the AWS API.

We unit tested the new metrics for postgres. We did not see the point of
testing the new cloudwatch ones, but they can be tested manually.

How to review?
-------------

 1. Code review
 2. Check if the metrics make sense
 3. Check if the naming of metrics and tags makes sense.

The SQL based metrics can be tested in a docker container locally.

The cloudwatch can be tested by running this locally as:

 1. edit `example-config.json` to point to a existing CF deployment with RDS instances.
 2. load your AWS creds.
 3. run `go run main.go -stdoutEmitter -config example_config.json`

Who?
----

Anyone but me
